kotlin-native/runtime/src/legacymm/cpp/Memory.cpp

```diff
@@ -252,7 +252,17 @@ class CycleDetector : private kotlin::Pinned, public KonanAllocatorAware {
 
 // TODO: can we pass this variable as an explicit argument?
 THREAD_LOCAL_VARIABLE MemoryState* memoryState = nullptr;
-THREAD_LOCAL_VARIABLE FrameOverlay* currentFrame = nullptr;
+
+#ifdef KONAN_WINDOWS
+    #include <processthreadsapi.h>
+    static DWORD currentFrameTlsIndex = TlsAlloc();
```

Comment:
Shouldn't we call `TlsFree` eventually?

Reply:
This is program-wide and I guess resources would be freed anyway. Is there a destructor where that could be added for example when generating a dll that is loaded and unloaded? Otherwise we could add an atexit function call, in the case this is needed.

Reply:
> This is program-wide

Is this true? One can load and unload a Kotlin-produced dll dynamically.

> Is there a destructor where that could be added for example when generating a dll that is loaded and unloaded?

Could you please investigate this?

> Otherwise we could add an atexit function call, in the case this is needed.

This doesn't seem to cover the case I described above.

Reply:
In theory, this should be fixed already in the newer commits. `atexit` seems to work when the DLL is unloaded. Check the spike sample I have provided if you want to verify that in isolation.

Reply:
> `atexit` seems to work when the DLL is unloaded.

Btw, is this documented?